src/lib/containers/entity/annotations/AdditionalPropertiesSchemaField.tsx

```diff
@@ -209,7 +222,10 @@ export function AdditionalPropertiesSchemaField<T>(
           placeholder={props.placeholder ?? ''}
           options={{}}
           formContext={props.formContext as T}
-          onBlur={props.onBlur}
+          onBlur={(id, value) => {
+            setList(transformDataFromPropertyType(list, propertyType))
+            props.onBlur(id, value)
+          }}
```

Comment:
Re-cast the value to its type on blur (e.g., if you have `7.5` in an integer field, we will convert it to `7` on blur).

Reply:
Like the floor truncation - it's simple and easy to predict the behavior.